include/hal/library/cryptlib/cryptlib_sm2.h

```diff
@@ -0,0 +1,268 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2021-2022 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#ifndef CRYPTLIB_SM2_H
+#define CRYPTLIB_SM2_H
+
+/*=====================================================================================
+ *    Shang-Mi2 Primitives
+ *=====================================================================================*/
+
+/**
+ * Allocates and Initializes one Shang-Mi2 context for subsequent use.
+ *
+ * The key is generated before the function returns.
```

Comment:
@jyao1 What does this mean? Can this comment be deleted?

Reply:
Yes, please